packages/live-preview-sdk/src/csm/index.ts

```diff
@@ -0,0 +1,74 @@
+import { vercelStegaEncode } from '@vercel/stega';
+import { debug } from '../helpers';
+import jsonPointer from 'json-pointer';
+import { EntitySource, GraphQLResponse, Source } from './types';
+
+export class ContentSourceMaps {
+  private isUrlOrIsoDate(value: string) {
+    // Regular expression for URL validation
+    const urlRegex = /^(http|https):\/\/[^ "]+$/;
+    // Regular expression for ISO 8601 date validation
+    const isoDateRegex = /^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}(.\d+)?([+-]\d{2}:\d{2}|Z)?$/;
+
+    // Check if the string matches URL or ISO 8601 date format
+    return urlRegex.test(value) || isoDateRegex.test(value);
+  }
+
+  private getHref(
+    source: Source,
+    entries: EntitySource[],
+    assets: EntitySource[],
+    spaces: string[],
+    environments: string[],
+    fields: string[],
+    locales: string[]
+  ): string | null {
+    const isEntry = 'entry' in source;
+    const entity = isEntry ? entries[source.entry] : assets[source.asset];
+    if (!entity) return null;
+
+    const space = spaces[entity.space];
+    const environment = environments[entity.environment];
+    const entityId = entity.id;
+    const field = fields[source.field];
+    const locale = locales[source.locale];
+
+    const basePath = `https://app.contentful.com/spaces/${space}/environments/${environment}`;
```

Comment:
Does this need to handle eu residency?

Reply:
Good call! 🇪🇺
Wonder how I would handle that. I guess somehow relying on the targetOrigin of ContentfulLivePreviewInitConfig? Or do we accept a second parameter for the encodeSourceMap function that has the targetOrigin?